### PR TITLE
Adds new method for determining option for default object rights

### LINKED ISSUE #2631: Edit APO form loads the incorrect default object rights in the menu when rights are location-based
**Describe the bug**
When the default object rights on an APO are location-based, the Edit APO form pre-populates the access rights dropdown with the value of "World".

APO https://argo-qa.stanford.edu/view/druid:vr838wn4827 has rights as location: Archive of Recorded Sound ("ars"):

Screen Shot 2021-04-02 at 1 53 51 PM

But when loading the Edit APO form, the menu shows "world" selected:
Screen Shot 2021-04-02 at 1 55 26 PM

**User Impact**
It would be easy to accidentally change the default object rights to world when editing the APO to change another value.

**Expected behavior**
The currently selected value should populate the menu.

### PR DESCRIPTION
**Pull request by jermnelson**
## Why was this change made?
Fixes #2631

## How was this change tested?
New unit tests

## Which documentation and/or configurations were updated?
n/a

### CODE REVIEW

File: app/forms/apo_form.rb [hunk 1]
```diff
@@ -79,6 +79,19 @@ def collection_radio
     'none'
   end
 
+  # rubocop:disable Metrics/CyclomaticComplexity
```
Comment (jermnelson) [inline app/forms/apo_form.rb:82]: Open to suggestions on how to refactor this method to remove these rubocop exceptions.

File: app/forms/apo_form.rb [hunk 2]
```diff
@@ -79,6 +79,19 @@ def collection_radio
...
+  # rubocop:disable Metrics/PerceivedComplexity
+  def default_object_rights
+    return default_rights if model.nil?
+    return "loc:#{model.administrative.defaultAccess.readLocation}" if default_rights == 'location-based'
+    return 'cdl-stanford-nd' if model.administrative&.defaultAccess&.controlledDigitalLending
```
Comment (jcoyne) [inline app/forms/apo_form.rb:87]: why do you need &. in this line but not on the prior line?

File: app/forms/apo_form.rb (hunk 1 above)
Comment (jcoyne) [reply app/forms/apo_form.rb:82]: how about doing something like `default_access = model.administrative.defaultAccess` and using that local var throughout. That will likely reduce complexity substantially.

Comment (jermnelson) [reply app/forms/apo_form.rb:82]: Thanks! That changed helped drop the complexity to 8/7. Moved some of the code to the private `check_default_object_rights` method and now passes.

File: app/forms/apo_form.rb (hunk 2 above)
Comment (jermnelson) [reply app/forms/apo_form.rb:87]: Thanks for the catch; updated in latest commit.

File: app/forms/apo_form.rb [hunk 3]
```diff
@@ -79,6 +79,14 @@ def collection_radio
     'none'
   end
 
+  def default_object_rights
+    return default_rights if model.nil?
+
+    default_object_rights_option = check_default_object_rights
```
Comment (jcoyne) [inline app/forms/apo_form.rb:85]: can you just return `check_default_object_rights || default_rights`?

File: app/forms/apo_form.rb [hunk 4]
```diff
@@ -101,6 +109,14 @@ def options_for_use_license_type(current_value)
     end.compact # the options block will produce nils for unused deprecated entries, compact will get rid of them
   end
 
+  def check_default_object_rights
+    default_access = model.administrative&.defaultAccess
```
Comment (jcoyne) [inline app/forms/apo_form.rb:113]: How about handling the case where default_access is `nil` first? Then you can get rid of the rest of the `&.` in this method.

File: app/forms/apo_form.rb (hunk 3 above)
Comment (jermnelson) [reply app/forms/apo_form.rb:85]: 👍

File: app/forms/apo_form.rb (hunk 4 above)
Comment (jermnelson) [reply app/forms/apo_form.rb:113]: Done, thanks!

File: app/forms/apo_form.rb (hunk 3 above)
Comment (jcoyne) [inline app/forms/apo_form.rb:85]: Do we need this allocation? Can you just move the call to the last line?

Comment (jermnelson) [reply app/forms/apo_form.rb]: Done

File: app/views/apo/_form.html.erb [hunk 5]
```diff
@@ -103,7 +103,7 @@
 
   <div class="form-group">
     <%= f.label :default_rights, "Default Object Rights" %>
-    <%= f.select :default_rights, options_for_select(Constants::REGISTRATION_RIGHTS_OPTIONS, @form.default_rights), {}, class: 'form-control' %>
+    <%= f.select :default_rights, options_for_select(Constants::REGISTRATION_RIGHTS_OPTIONS, @form.default_object_rights), {}, class: 'form-control' %>
```
Comment (jcoyne) [inline app/views/apo/_form.html.erb:106]: It's conventional for the existing value to come from a method with the same name as the form field (as they were previously).  Can you align these?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/views/apo/_form.html.erb (hunk 5 above)
Comment (jcoyne) [reply app/views/apo/_form.html.erb:106]: I think you might want to keep the ERB as it was and change this line: https://github.com/sul-dlss/argo/blob/48b8c859a723eba4092f50870c6de1d6e2560ffc/app/forms/apo_form.rb#L38 to call the `default_object_rights` method you defined.

Comment (jermnelson) [reply app/views/apo/_form.html.erb]: Done, also made `default_object_rights` private and adjusted the tests accordingly

**Review by jcoyne — APPROVED** [review]

**Pull request merged**